File: CSharpFundamentals/14 Strings_Dictionaries_LINQ/CensorYourEmail/CensorYourEmail.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CensorYourEmail
{
    class CensorYourEmail
    {
        static void Main(string[] args)
        {
            string email = Console.ReadLine();
            var text = Console.ReadLine().Split(' ').ToArray();
            string username = email.Split('@')[0];
            string domain = email.Split('@')[1];
            
            string mail = new string('*', username.Length) + "@" + domain;
            for (int i = 0; i < text.Length; i++)
            {
                if (text[i] == email)
                    text[i] = mail; 
            }
            Console.WriteLine(string.Join(" ", text));
        }
    }
}

File: SoftwareTechnologies/Java/JavaBasics/src/CensorYourEmail.java
import java.util.Arrays;
import java.util.Scanner;

public class CensorYourEmail {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        String email = scan.nextLine();

        String[] text = scan.nextLine().split(" ");

        String username = email.split("@")[0];
        String domain = email.split("@")[1];
        String stars = "";
        for (int i = 0; i < username.length(); i++) {
            stars+="*";
        }
        String mail = stars + "@" + domain;

        for (int i = 0; i < text.length; i++)
        {
            if (text[i].equals(email))
                text[i] = mail;
        }
        System.out.print(String.join(" ", text));
    }
}
